File: Emplyfi/ems-backend/src/main/java/com/ems/ems_backend/service/DepartmentService.java
package com.ems.ems_backend.service;

import com.ems.ems_backend.dto.DepartmentDto;

public interface DepartmentService {

	// create
	public DepartmentDto createDepartment(DepartmentDto departmentDto);

	// update
	public DepartmentDto updateDepartment(Long id, DepartmentDto updateDepartment);

	// delete
	public void deleteDepartment(Long id);
}

File: Emplyfi/ems-frontend/src/services/DepartmentService.js
import axios from "axios";

const BASE_URL = 'http://localhost:8080/departments';

export const listDepartments = () => axios.get(`${BASE_URL}/getAllDepartments`);

export const createDepartment = (department) => axios.post(`${BASE_URL}/create`, department);

export const getDepartmentById = (departmentId) => axios.get(`${BASE_URL}/getDepartmentById/${departmentId}`);

export const updateDepartment = (departmentId, department) => axios.put(`${BASE_URL}/updateDepartment/${departmentId}`, department);

export const deleteDepartment = (departmentId) => axios.delete(`${BASE_URL}/deleteDepartment/${departmentId}`);
